supervisor/shared/web_workflow/web_workflow.c

```diff
@@ -236,21 +240,46 @@ void supervisor_start_web_workflow(void) {
         return;
     }
 
-    mdns_server_construct(&mdns, true);
-    common_hal_mdns_server_set_instance_name(&mdns, MICROPY_HW_BOARD_NAME);
-    common_hal_mdns_server_advertise_service(&mdns, "_circuitpython", "_tcp", 80);
+    char port_encoded[6];
+    size_t port_len = 0;
+    size_t new_port = web_api_port;
+    #if CIRCUITPY_DOTENV
+    port_len = dotenv_get_key("/.env", "CIRCUITPY_WEB_API_PORT", port_encoded, sizeof(port_encoded) - 1);
+    #endif
+    if (0 < port_len && port_len < sizeof(port_encoded)) {
```

Comment:
Is there a `port_encoded[5] = 0` or `port_encoded[port_len] = 0` needed here, to ensure the string is terminated before passing it to strtoul? (since port_encoded is a local variable with no initializer, I think the contents are unpredictable)